### PR TITLE
GDScript crash on builtin type constructor fix

### LINKED ISSUE #41705: Crash caused by dynamic-stack-buffer-overflow when running project after recent GDScript changes
**Godot version:**
Godot 4.0 358e209fa049887fd4212c52c9faba0ca9d6e111

**OS/device including version:**
Ubuntu 20.04

**Issue description:**
This happens after recent GDScript changes(after merging GDScript 2.0)
Address sanitizer log
```
SUMMARY: AddressSanitizer: dynamic-stack-buffer-overflow /mnt/Miecz/mojgodot/modules/gdscript/gdscript_function.cpp:953:17 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*)
Shadow bytes around the buggy address:
  0x10002aac2510: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10002aac2520: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10002aac2530: 00 00 00 00 00 00 00 00 ca ca ca ca 00 00 00 00
  0x10002aac2540: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10002aac2550: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x10002aac2560: 00[cb]cb cb cb cb cb cb f1 f1 f1 f1 00 f2 f2 f2
  0x10002aac2570: 00 f2 f2 f2 00 00 00 f2 f2 f2 f2 f2 00 00 00 f2
  0x10002aac2580: f2 f2 f2 f2 00 00 00 f2 f2 f2 f2 f2 04 f2 04 f2
  0x10002aac2590: 00 00 00 f2 f2 f2 f2 f2 f8 f8 f8 f2 f2 f2 f2 f2
  0x10002aac25a0: f8 f2 f2 f2 f8 f8 f8 f2 f2 f2 f2 f2 f8 f8 f8 f2
  0x10002aac25b0: f2 f2 f2 f2 00 f2 f2 f2 01 f2 04 f2 00 00 00 f2
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==77091==ABORTING

```

**Steps to reproduce:**
1. Run project

**Minimal reproduction project:**
https://github.com/qarmin/RegressionTestProject/archive/master.zip

### LINKED ISSUE #41848: GDScript: Invalid method name in callable constructor cause a crash
**Godot version:** adc931fae

**OS/device including version:**  windows10

**Issue description:**
running the below code cause a crash.
```gdscript
extends Node2D

func _ready():
	var callable = Callable(self, 'blah blah...')

```
(backtrace)
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] GDScriptFunction::call (C:\dev\godot\modules\gdscript\gdscript_function.cpp:953)
[1] GDScriptFunction::call (C:\dev\godot\modules\gdscript\gdscript_function.cpp:953)
[2] GDScriptInstance::call (C:\dev\godot\modules\gdscript\gdscript.cpp:1317)
[3] ScriptInstance::call (C:\dev\godot\core\script_language.cpp:308)
[4] Node::_notification (C:\dev\godot\scene\main\node.cpp:147)
[5] Node::_notificationv (C:\dev\godot\scene\main\node.h:46)
[6] CanvasItem::_notificationv (C:\dev\godot\scene\main\canvas_item.h:162)
[7] Node2D::_notificationv (C:\dev\godot\scene\2d\node_2d.h:37)
[8] Object::notification (C:\dev\godot\core\object.cpp:808)
[9] Node::_propagate_ready (C:\dev\godot\scene\main\node.cpp:189)
[10] Node::_propagate_ready (C:\dev\godot\scene\main\node.cpp:181)
[11] Node::_set_tree (C:\dev\godot\scene\main\node.cpp:2564)
[12] SceneTree::init (C:\dev\godot\scene\main\scene_tree.cpp:396)
[13] OS_Windows::run (C:\dev\godot\platform\windows\os_windows.cpp:625)
[14] widechar_main (C:\dev\godot\platform\windows\godot_windows.cpp:163)
[15] _main (C:\dev\godot\platform\windows\godot_windows.cpp:185)
[16] main (C:\dev\godot\platform\windows\godot_windows.cpp:199)
[17] WinMain (C:\dev\godot\platform\windows\godot_windows.cpp:213)
[18] __scrt_common_main_seh (d:\A01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[19] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### LINKED ISSUE #42024: [4.0] crash when constructing a Vector3 with the delta parameter from _process(delta)
**Godot version:**
Godot Engine v4.0.dev.custom_build.43725691b

**OS/device including version:**
Windows 10, Vulkan

**Issue description:**
When constructing a Vector3 inside the _process function with the delta parameter, the game crashes.

**Steps to reproduce:**
```gdscript
func _process(delta):
	var movement = Vector3(0, 0, delta)
```
I also tried explicitly making the zeros into floats, but there is still a crash:
```gdscript
func _process(delta):
	var movement = Vector3(0.0, 0.0,delta)
```

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/5213942/test.zip)

### LINKED ISSUE #42156: crash by var pos=Vector3(1.0,1.0,n)
Godot version:4.0
OS/device including version:mac os 10.15.6

Issue description: Crash

Steps to reproduce:

`
extends Node3D
func _ready():
	var n=10.0
	var pos=Vector3(1.0,1.0,n)
`

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #41848
Fix: #42156
Fix: #42024 
Fix: #41705

### CODE REVIEW

File: modules/gdscript/gdscript_byte_codegen.cpp [hunk 1]
```diff
@@ -529,6 +529,7 @@ void GDScriptByteCodeGenerator::write_construct(const Address &p_target, Variant
 		append(p_arguments[i]);
 	}
 	append(p_target);
+	alloc_call(p_arguments.size());
```
Comment (akien-mga) [inline modules/gdscript/gdscript_byte_codegen.cpp:532]: Would there be the same issue with `write_construct_array` and `write_construct_dictionary`?

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_byte_codegen.cpp:532]: @akien-mga `Array` and `Dictionary` are constructed locally but other types are constructed by calling the constructor(`Variant::construct()`) at ` GDScriptFunction::call()`  which seems inconsistent?

**Comment by DavidSichma** [comment]
Also fixes https://github.com/godotengine/godot/issues/42024

**Comment by Arakash** [comment]
nice, I've been waiting for this fix. Thanks!

**Comment by ThakeeNathees** [comment]
@vnen could you review this
(I've exams next week so I might be slow to respond :) )

**Review by vnen — APPROVED** [review]
Sorry for the slow response.

**Comment by vnen** [comment]
Thanks!

**Pull request merged**